spec/active_record/connection_adapters/oracle_enhanced_adapter_spec.rb

```diff
@@ -729,4 +729,29 @@ class TestLog < ActiveRecord::Base
       expect(TestLog.count).to eq 2
     end
   end
+
+  describe "synonym_names" do
+    before(:all) do
+      schema_define do
+        create_table :test_comments, force: true do |t|
+          t.string :comment
+        end
+        add_synonym :synonym_comments, :test_comments
+      end
+    end
+
+    after(:all) do
+      schema_define do
+        drop_table :test_comments
+        remove_synonym :synonym_comments
+      end
+      ActiveRecord::Base.clear_cache! if ActiveRecord::Base.respond_to?(:"clear_cache!")
+    end
+
+    it "includes synonyms in data_source" do
+      conn = ActiveRecord::Base.connection
+      expect(conn).to be_data_source_exist("synonym_comments")
```

Comment:
Looks like both `be_data_source_exist` and `be_data_source_exists` are working.  May be this is a common knowledge between RSpec users, but I just do not know.